disease_gene/disease_associates_gene/dataset_statistics/dataset_statistics.py

```diff
@@ -0,0 +1,491 @@
+
+
+# In[29]:
+
+
+dev_candidates_stats_df = (
+    filtered_total_candidates_df
+    .query("split==1")
+    .merge(dev_dataset_df[["candidate_id", "curated_dsh"]], on="candidate_id")
+)
+print("Total number of positives (1) and negatives (0): \n{}".format(dev_candidates_stats_df.curated_dsh.value_counts()))
+
+
+# In[30]:
+
+
+ax = sns.scatterplot(
+    x="Disease_mention_count", 
+    y="Gene_mention_count", 
+    data=
+    (
+        dev_candidates_stats_df
+        .drop_duplicates("sentence_id")
+        .groupby(["Disease_mention_count", "Gene_mention_count"]).size()
+        .reset_index()
+        .rename(index=str, columns={0:"size"})
+        .assign(natural_log_size= lambda x: pd.np.log(x['size']))
+    ),
+    hue="natural_log_size",
+    palette='viridis'
+)
+ax.set_title("Sentence Distribution of Hand Labeled Dev Dataset")
+
+
+# In[31]:
+
+
+sns.distplot(dev_candidates_stats_df["sen_length"], rug=False)
+
+
+# ## Test Set
+
+# In[32]:
+
+
+ax = sns.scatterplot(
+    x="Disease_mention_count", 
+    y="Gene_mention_count", 
+    data=
+    (
+        filtered_total_candidates_df
+        .query("split==2")
+        .drop_duplicates("sentence_id")
+        .groupby(["Disease_mention_count", "Gene_mention_count"]).size()
+        .reset_index()
+        .rename(index=str, columns={0:"size"})
+        .assign(natural_log_size= lambda x: pd.np.log(x['size']))
+    ),
+    hue="natural_log_size",
+    palette='viridis'
+)
+ax.set_title("Sentence Distribution of Total Test Dataset")
+
+
+# In[33]:
+
+
+test_dataset_df = pd.read_excel("sentence_labels_test.xlsx").query("curated_dsh.notnull()")
+
+
+# In[34]:
+
+
+test_candidates_stats_df = (
+    filtered_total_candidates_df
+    .query("split==2")
+    .merge(test_dataset_df[["candidate_id", "curated_dsh"]], on="candidate_id")
+)
+print("Total number of positives (1) and negatives (0): \n{}".format(test_candidates_stats_df.curated_dsh.value_counts()))
+
+
+# In[35]:
+
+
+ax = sns.scatterplot(
+    x="Disease_mention_count", 
+    y="Gene_mention_count", 
+    data=
+    (
+        test_candidates_stats_df
+        .drop_duplicates("sentence_id")
+        .groupby(["Disease_mention_count", "Gene_mention_count"]).size()
+        .reset_index()
+        .rename(index=str, columns={0:"size"})
+        .assign(natural_log_size= lambda x: pd.np.log(x['size']))
+    ),
+    hue="natural_log_size",
+    palette='viridis'
+)
+ax.set_title("Sentence Distribution of Hand Labeled Dev Dataset")
+
+
+# In[36]:
+
+
+sns.distplot(test_candidates_stats_df["sen_length"], rug=False)
+
+
+# In conclusion, it is important to analyze one's dataset before undergoing the journey of training deep learning model.  Sentence length is an important cutoff, when it comes to filtering the dataset; otherwise, a significant amount of problems will arise when it comes to training machine learning algorithms. Optimal cutoff point in this case would be: **84** tokens/words or smaller depending on algorithm performance.
```

Comment:
Why does an outlier in the sentence length affect the learning so much?  i.e. How is sentence length being used by the training algorithm?